src/new-components/banner/banner.tsx

```diff
@@ -0,0 +1,101 @@
+import * as React from 'react'
+import { Box } from '../box'
+import { Columns, Column } from '../columns'
+import { useId } from '../common-helpers'
+
+import styles from './banner.module.css'
+
+export type BannerTone = 'info' | 'promotion'
+
+type BannerProps = {
+    id?: string
+    /**
+     * The tone of the Banner. Affects the background color and the outline.
+     */
+    tone: BannerTone
+    /**
+     * The icon that should be added inside the Banner.
+     */
+    icon: React.ReactChild
+    /**
+     * The title to be displayed at the top of the Banner.
+     */
+    title: React.ReactNode
+    /**
+     * An optional description to be displayed inside the Banner.
+     */
+    description?: React.ReactNode
+    /**
+     * An optional action to displayed inside the Banner.
+     */
+    action?: React.ReactChild
+}
+
+const Banner = React.forwardRef<HTMLDivElement, BannerProps>(function Banner(
+    { id, tone, icon, title, description, action, ...props }: BannerProps,
+    ref,
+) {
+    const titleId = useId()
+    const descriptionId = useId()
+    return (
+        <Box
+            {...props}
+            ref={ref}
+            id={id}
+            role="status"
+            aria-labelledby={titleId}
+            aria-describedby={descriptionId}
+            aria-live="polite"
+            tabIndex={0}
+            borderRadius="standard"
+            className={[styles.banner, styles[`banner-${tone}`]]}
+        >
+            <Columns space="medium" alignY="center">
+                <Column
+                    width="content"
+                    aria-hidden
+                    style={{
+                        /* Make sure the icon is centered vertically */
+                        lineHeight: 0,
```

Comment:
This comes from the existing banner-like component we implemented in todoist-web https://github.com/Doist/todoist-web/blob/main/src/react_components/project-view/project-view-header.tsx#L349-L353.

I didn't have this at first and the icons were placed slightly off center, which was somewhat noticeable for banners without description (only title).